drivers/usb/udc/udc_stm32.c

```diff
@@ -230,7 +238,11 @@ static int udc_stm32_tx(const struct device *dev, uint8_t ep,
 		/* Wait for an empty package from the host.
 		 * This also flushes the TX FIFO to the host.
 		 */
-		usbd_ctrl_feed_dout(dev, 0);
+		if (DT_HAS_COMPAT_STATUS_OKAY(st_stm32_usb)) {
```

Comment:
That is not future-proof, right way would be to store controller type in dev->config and check it here.

Reply:
Should the driver support multiple instance at some point?

Reply:
I did not ask for changes, I just made a comment and left it up to you. And I did something similar some time ago, where the drivers were a bit simpler, but still not able to support different instances.